script.js

```diff
@@ -2,71 +2,78 @@ if (localStorage.getItem('data') === null) {
   localStorage.setItem('data', '[]');
 }
 
-let bookList = [];
+class BookList {
+  constructor() {
+    this.bookTemplate = {};
+    this.books = [];
+  }
 
-function Book(id, title, author) {
-  this.id = id;
-  this.title = title;
-  this.author = author;
-}
-function addBooks(bookList) {
-  const books = document.getElementById('books');
-  const liToRemove = document.querySelectorAll('li');
-  liToRemove.forEach((item) => {
-    item.remove();
-  });
+  book(title, author) {
+    this.bookTemplate.id = Date.now();
+    this.bookTemplate.title = title;
+    this.bookTemplate.author = author;
+    return this.bookTemplate;
+  }
 
-  bookList.forEach((book) => {
-    const li = document.createElement('li');
-    const hr = document.createElement('hr');
-    const removeBtn = document.createElement('input');
-    removeBtn.id = book.id;
-    removeBtn.className = 'remove-btn';
-    removeBtn.setAttribute('onclick', `removeBook( ${book.id} )`);
-    removeBtn.setAttribute('type', 'button');
-    removeBtn.setAttribute('value', 'Remove');
-    const titleBlock = document.createElement('div');
-    const title = document.createTextNode(book.title);
-    titleBlock.appendChild(title);
-    const authorBlock = document.createElement('div');
-    const author = document.createTextNode(book.author);
-    authorBlock.appendChild(author);
+  add(title, author) {
+    this.books.push(this.book(title, author));
+    this.save();
+  }
 
-    li.appendChild(titleBlock);
-    li.appendChild(authorBlock);
-    li.appendChild(removeBtn);
-    li.appendChild(hr);
-    books.appendChild(li);
-  });
-}
+  remove(id) {
+    this.books = this.books.filter((book) => book.id !== Number(id));
+    this.save();
+  }
 
-function getFromLocalStorage() {
-  return JSON.parse(localStorage.getItem('data'));
-}
+  save() {
+    localStorage.setItem('data', JSON.stringify(this.books));
+    this.updateView();
+  }
+
+  retrieve() {
+    this.books = JSON.parse(localStorage.getItem('data'));
+  }
```

Comment:
- [ ] Please try to avoid using the local storage inside the class method. Let's try to fix this by getting the books from a getter method of class and then storing the `data` in local storage if required.